datadog-crashtracker/src/crash_info/telemetry.rs

```diff
@@ -107,13 +107,45 @@ impl CrashPing {
         &self.message
     }
 
-    pub fn endpoint(&self) -> &Option<Endpoint> {
-        &self.endpoint
+    pub fn metadata(&self) -> &Metadata {
+        &self.metadata
+    }
+
+    pub fn siginfo(&self) -> &SigInfo {
+        &self.siginfo
     }
 
     fn current_schema_version() -> String {
         "1.0".to_string()
     }
+
+    pub fn upload_to_endpoint(&self, endpoint: &Option<Endpoint>) -> anyhow::Result<()> {
+        let rt = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
```

Comment:
can we check whether the endpoint is file here too? to avoid creating tokio runtime in that case?